### DISCUSSION TITLE
Best Approach to Revealing Objects

### DISCUSSION THREAD

**Discussion by 79pxls** [google/model-viewer#4205] [Q&A]
Dear community,
I'm looking for some advice for the best approach to revealing objects within model viewer.

I have a chassis with an EV battery and EV motor. These 2 parts come with 2 options. The model will load with option 1 for both parts visible. The user can toggle the battery, and or motor, to upgraded versions that will become visible when the UI buttons are clicked. The user can go back and forth with the selections and in any combination.

**Buttons**
[Battery V1]  [Battery V2]  [Motor V1]  [Motor V2]

Would it be best to handle this with a series of animations that are triggered from the buttons?
Is it possible to animate the shaders Alpha Factor with Blend enabled to fade the parts in and out?
Are there any tricks that accomplish the goal easily?

I'm not great at coding within model viewer so please be gentle.
**Thanks** in advance!

**Comment by elalish** [comment]
Sounds like you may have found an answer in #4168? I really do need to write up an example of this. If you happen to have a GLB authored with separate materials for the separate parts that you'd be willing to let us have under an open license, that would help a lot. We'll give credit, of course.

**Reply by 79pxls** [reply]
The suggestion in #4168 is very useful but not what I'm looking for.

It sounds like you are recommending the "animate the material Alpha Factor with Blend enabled" method to achieve the effect.
I can't share the model I'm working with due to an NDA but I could put something else together and send it to you with a list of the shader names.

Just to make sure...I can rename the materials in the [Google Model Viewer Editor](https://modelviewer.dev/editor/)?
Should all the Alpha Factors be set to 1 and you handle it in the code or should I make the hidden parts 0 and just give you the material names?

Thanks for the help @elalish !!

**Reply by elalish** [reply]
That would be great. Probably set alpha = 1 for the default "variant" and the others to zero, then let me know which combinations are interesting, so I can change those values in JS.
